scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,227 @@
+   * @return file name pattern is correct or not
+   */
+  private fun checkProhibitedFileNamePattern(
+    repoPath: String,
+    searchFiles: List<File>,
+    filenameCheck: FilenameCheck,
+  ): Boolean {
+    val prohibitedFilenameRegex = filenameCheck.getProhibitedFilenameRegex().toRegex()
+
+    val matchedFiles = searchFiles.filter { file ->
+      return@filter file.name !in filenameCheck.getExemptedFileNameList() &&
+        prohibitedFilenameRegex.matches(
+          RepoFile.retrieveFilePath(
+            file,
+            repoPath
+          )
+        )
+    }
+
+    logProhibitedFilenameFailure(
+      repoPath,
+      filenameCheck.getFailureMessage(),
+      matchedFiles,
+    )
+    return matchedFiles.toList().size != 0
+  }
+
+  /**
+   * Checks for a prohibited file content.
+   *
+   * @param repoPath the path of the repo
+   * @param searchFiles a list of all the files which needs to be checked
+   * @param fileContentCheck proto object of FileContentCheck
+   * @return file content pattern is correct or not
+   */
+  private fun checkProhibitedContent(
+    repoPath: String,
+    searchFiles: List<File>,
+    fileContentCheck: FileContentCheck,
+  ): Boolean {
+    val fileNameRegex = fileContentCheck.getFilenameRegex().toRegex()
+
+    val prohibitedContentRegex =
+      fileContentCheck.getProhibitedContentRegex().toRegex()
+
+    val matchedFiles = searchFiles.filter { file ->
+      file.name !in fileContentCheck.getExemptedFileNameList() &&
+        fileNameRegex.matches(file.name) &&
+        File(file.toString()).bufferedReader().lineSequence()
+          .foldIndexed(initial = false) { lineIndex, isFailing, lineContent ->
+            val matches = prohibitedContentRegex.matches(lineContent)
+            if (matches) {
+              logProhibitedContentFailure(
+                // Since, the line number starts from 1 and index starts
+                // from 0, therefore we have to increment index by 1, to
+                // denote the line number.
+                lineIndex + 1,
+                fileContentCheck.getFailureMessage(),
+                RepoFile.retrieveFilePath(file, repoPath)
+              )
+            }
+            isFailing || matches
+          }
+    }
+    return matchedFiles.size != 0
+  }
+
+  /**
+   * Logs the failures for filename pattern violation.
+   *
+   * @param repoPath the path of the repo to be analyzed
+   * @param errorToShow the filename error to be logged
+   * @param matchedFiles a list of all the files which had the filenaming violation
+   * @return log the filename failures
+   */
+  private fun logProhibitedFilenameFailure(
+    repoPath: String,
+    errorToShow: String,
+    matchedFiles: List<File>
+  ) {
+    if (matchedFiles.size != 0) {
+      println("Filename pattern violation: $errorToShow")
+      matchedFiles.forEach {
+        println("Prohibited file: [ROOT]/${RepoFile.retrieveFilePath(it, repoPath)}")
```

Comment:
You could instead say ``//`` instead of [ROOT] since that's the Bazel way to reference the root (& people should become more familiar with using it over time).

Reply:
We are logging the absolute path now, as suggested latest.